packages/react-native/React/Views/UIView+React.m

```diff
@@ -242,31 +245,33 @@ - (void)setReactTransformOrigin:(RCTTransformOrigin)reactTransformOrigin
 
 static void updateTransform(UIView *view)
 {
-  CGSize size = view.bounds.size;
-  RCTTransformOrigin transformOrigin = view.reactTransformOrigin;
-
-  CGFloat anchorPointX = 0;
-  CGFloat anchorPointY = 0;
-  CGFloat anchorPointZ = 0;
-
-  if (transformOrigin.x.unit == YGUnitPoint) {
-    anchorPointX = transformOrigin.x.value - size.width * 0.5;
-  } else if (transformOrigin.x.unit == YGUnitPercent) {
-    anchorPointX = (transformOrigin.x.value * 0.01 - 0.5) * size.width;
-  }
-
-  if (transformOrigin.y.unit == YGUnitPoint) {
-    anchorPointY = transformOrigin.y.value - size.height * 0.5;
-  } else if (transformOrigin.y.unit == YGUnitPercent) {
-    anchorPointY = (transformOrigin.y.value * 0.01 - 0.5) * size.height;
+  CATransform3D transform;
+  id rawTansformOrigin = objc_getAssociatedObject(view, @selector(reactTransformOrigin));
+  if (rawTansformOrigin) {
+    CGSize size = view.bounds.size;
+    CGFloat anchorPointX = 0;
+    CGFloat anchorPointY = 0;
+    CGFloat anchorPointZ = 0;
+    RCTTransformOrigin transformOrigin = view.reactTransformOrigin;
```

Comment:
Let's avoid another lookup of the associated object here (we already have one in `reactSetFrame`), and duplicate the `getValue`.

```suggestion
    RCTTransformOrigin transformOrigin;
    [rawTansformOrigin getValue:&transformOrigin];
```

Reply:
Makes sense. Updated!